pkg/core/policy/dataplane_matcher.go

```diff
@@ -93,7 +93,7 @@ func SelectInboundDataplanePolicies(dataplane *mesh.DataplaneResource, policies
 					continue
 				}
 				tagSelector := mesh_proto.TagSelector(selector.Match)
-				if inbound.MatchTags(tagSelector) {
+				if tagSelector.Matches(inbound.Tags) {
```

Comment:
I suppose this is the right way around but I find it very difficult to know which way is which...

Reply:
Yeah I've removed some of the helper functions to make this less confusing but it's still 2 maps...